src/main/java/com/example/SS2_Backend/ss/smt/implement/TripletOTOProblem.java

```diff
@@ -0,0 +1,291 @@
+import com.example.SS2_Backend.util.StringUtils;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.moeaframework.core.Solution;
+import org.moeaframework.core.Variable;
+import org.moeaframework.core.variable.Permutation;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+@Slf4j
+@Getter
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@AllArgsConstructor
+public class TripletOTOProblem implements MatchingProblem {
+
+    /** problem name */
+    final String problemName;
+
+    /** problem size (number of individuals in matching problem */
+    final int problemSize;
+
+    /** number of set in matching problem */
+    final int setNum;
+
+    /** Matching data */
+    final MatchingData matchingData;
+
+    /** preference list  */
+    final PreferenceListWrapper preferenceLists;
+
+    /** problem fitness function */
+    final String fitnessFunction;
+
+    /** fitness evaluator */
+    final FitnessEvaluator fitnessEvaluator;
+
+    /** will not be used */
+    final int UNUSED_VAL = MatchingConst.UNUSED_VALUE;
+
+
+
+    /**
+     * generate new solution
+     * @return Solution contains Variable(s)
+     */
+    @Override
+    public Solution newSolution() {
+        Solution solution = new Solution(1, 1);
+        Permutation permutationVar = new Permutation(problemSize);
+        solution.setVariable(0, permutationVar);
+        return solution;
+    }
+
+    /**
+     * evaluate function for matching problem
+     * @param solution Solution contains Variable(s)
+     */
+    @Override
+    public void evaluate(Solution solution) {
+        Matches result = this.stableMatching(solution.getVariable(0));
+        // Check Exclude Pairs
+        int[][] excludedPairs = this.matchingData.getExcludedPairs();
+        if (Objects.nonNull(excludedPairs)) {
+            for (int[] excludedPair : excludedPairs) {
+                if (result.getSetOf(excludedPair[0]).contains(excludedPair[1])) {
+                    solution.setObjective(0, Double.MAX_VALUE);
+                    return;
+                }
+            }
+        }
+        double[] satisfactions = this.preferenceLists.getMatchesSatisfactions(result, matchingData);
+        double fitnessScore;
+        if (this.hasFitnessFunc()) {
+            fitnessScore = fitnessEvaluator
+                    .withFitnessFunctionEvaluation(satisfactions, this.fitnessFunction);
+        } else {
+            fitnessScore = fitnessEvaluator.defaultFitnessEvaluation(satisfactions);
+        }
+        solution.setAttribute(MatchingConst.MATCHES_KEY, result);
+        solution.setObjective(0, -fitnessScore);
+    }
+
+
+    /**
+     * check exists fitness function
+     * @return true if exists
+     */
+    public boolean hasFitnessFunc() {
+        return StringUtils.isEmptyOrNull(this.fitnessFunction);
```

Comment:
Đổi thành !StringUtils.isEmptyOrNull(this.fitnessFunction) && this.fitnessFunction.equalsIgnoreCase("default"); nhé